pkg/apis/core/validation/validation.go

```diff
@@ -4754,7 +4754,7 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 			case mungedPodSpec.Affinity == nil && oldNodeAffinity != nil:
 				mungedPodSpec.Affinity = &core.Affinity{NodeAffinity: oldNodeAffinity} // +k8s:verify-mutation:reason=clone
 			default:
-				mungedPodSpec.Affinity.NodeAffinity = oldNodeAffinity // +k8s:verify-mutation:reason=clone
+				mungedPodSpec.Affinity = oldPod.Spec.Affinity // +k8s:verify-mutation:reason=clone
```

Comment:
This is not correct because it will also copy the pod affinity fields, which are not mutable; we need to set mungedPodSpec.Affinity to nil if all the fields are nil.

Reply:
Hi @ahg-g, thanks for the review. Can you PTAL?